consensus/state_sync_handler.go

```diff
@@ -37,20 +37,23 @@ func (m *consensusModule) handleStateSyncMessage(stateSyncMessage *typesCons.Sta
 	switch stateSyncMessage.Message.(type) {
 	case *typesCons.StateSyncMessage_MetadataReq:
 		m.logger.Info().Str("proto_type", "MetadataRequest").Msg("Handling StateSyncMessage MetadataReq")
-		if !m.stateSync.IsServerModEnabled() {
+		if !m.serverMode {
 			return fmt.Errorf("server module is not enabled")
 		}
 		return m.stateSync.HandleStateSyncMetadataRequest(stateSyncMessage.GetMetadataReq())
 	case *typesCons.StateSyncMessage_MetadataRes:
-		return m.stateSync.HandleStateSyncMetadataResponse(stateSyncMessage.GetMetadataRes())
+		m.metadataReceived <- stateSyncMessage.GetMetadataRes()
+		return nil
 	case *typesCons.StateSyncMessage_GetBlockReq:
 		m.logger.Info().Str("proto_type", "GetBlockRequest").Msg("Handling StateSyncMessage MetadataReq")
-		if !m.stateSync.IsServerModEnabled() {
+		if !m.serverMode {
 			return fmt.Errorf("server module is not enabled")
 		}
 		return m.stateSync.HandleGetBlockRequest(stateSyncMessage.GetGetBlockReq())
 	case *typesCons.StateSyncMessage_GetBlockRes:
-		return m.stateSync.HandleGetBlockResponse(stateSyncMessage.GetGetBlockRes())
+		fmt.Println("Received block: ", stateSyncMessage.GetGetBlockRes().Block)
+		m.blocksReceived <- stateSyncMessage.GetGetBlockRes().Block
```

Comment:
In `metadataReceived`, you store response protobufs. In `blockReceived`, you store the data inside the block.

This is inconsistent. Doesn't matter what you do, but be consistent.

I think you should store the whole response in both, and let the actual logic decide what to do with it.